CONTRIBUTING.md

```diff
@@ -318,6 +320,92 @@ should be taken. If however, a code-owner is demonstrably shown to intentionally
 have had acted maliciously or grossly negligent, code-owner privileges may be
 stripped with no prior warning or consent from the member in question.
 
+Other potential removal criteria:
+  * Missing 3 scheduled meetings results in ICF evaluating whether the member should be
+    removed / replaced
+  * Violation of Code of Conduct
+  
 Earning this privilege should be considered to be no small feat and is by no
 means guaranteed by any quantifiable metric. It is a symbol of great trust of
 the community of this project.
+
+
+## Concept & Release Approval Process
+
+The process for how Cosmos SDK maintainers take features and ADRs from concept to release
+is broken up into three distinct stages: **Strategy Discovery**, **Concept Approval**, and 
+**Implementation & Release Approval**
+
+
+### Strategy Discovery
+
+* Develop long term priorities, strategy and roadmap for the SDK
+* Release committee not yet defined as there is already a roadmap that can be used for the time being
+
+### Concept Approval
+
+* Architecture Decision Records (ADRs) may be proposed by any contributors or maintainers of the Cosmos SDK,
+    and should follow the guidelines outlined in the
+    [ADR Creation Process](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/PROCESS.md)
+* After proposal, a time bound period for Request for Comment (RFC) on ADRs commences
+* ADRs are intended to be iterative, and may be merged into `master` while still in a `Proposed` status
+
+**Time Bound Period**
+
+* Time bound period for individual ADR Pull Requests should not exceed 2 weeks
+* Total time bound period for an ADR to reach a decision (`ABANDONED | ACCEPTED | REJECTED`) should not exceed
+    8 weeks
```

Comment:
> DR Pull Requests should not exceed 2 weeks

It's good to cap it. We already agreed on an iterative approach. Capping it will mobilize us to have an attention. 

> Total time bound [...] 8 weeks

I think can depend on the ADR scope - some can be a research intensive and can start with an idea and develop further. Maybe we should use a _should_ word here?

Reply:
Good points.

> It's good to cap it

In line of principle, I fully agree. Yet 2 weeks won't likely be enough for most ADR conversations. Let's not forget that ADRs are about architectural changes. What if the 2 weeks timeout expires and the ADR committee couldn't reach consensus? Maybe we want a mechanism to grant ADRs some extra time for discussion? Such mechanism could kick in in case further clarifications are required even if the ADR is in early stages.

However, this can go in as is for the time being. Looks like an excellent start :+1:

Reply:
2 weeks is only intended to be the time period between individual merges of open PRs. Since we are moving towards ADRs being more iterative, and not spending long amounts of time in an "open PR" status, I think this should be fine.

I just updated the language and dates slightly:
- 1 week for initial review by reviewers
- 2 weeks to merge the PR (in draft / proposed state is fine)
- 4 weeks to conclusion of ADR

After 4 weeks, the ADR committee is responsible for granting an extension, and setting a new time bound period if so desired.

Reply:
For research intensive ADRs, 4 weeks is not enough to go from start to [ABANDONED | ACCEPTED | REJECTED]. 
If committee will be lean and active to grant extensions then fine.

Reply:
> 1 week for initial review by reviewers
> 2 weeks to merge the PR (in draft / proposed state is fine)
> 4 weeks to conclusion of ADR

@clevinson this seems to be shorted than the original 8 weeks? How about the following:

```patch
- 1 week for initial review by reviewers
+ 2 weeks for initial review by reviewers
```

Reply:
I'm confused. Are you taking issue with 8 weeks total time, or with 1 week initial review?

1 week initial review I think is actually very reasonable. Since we have a 2 week cycle for architecture review calls, this will allow a proposer to get initial feedback in written form and still have time to respond to feedback before collectively discussing at a meeting. On meetings, everyone should have reviewed the ADR already, and a 1 week initial review seems like it will allow for that. 

The 8 weeks was never agreed upon, it was a number I proposed as a guesstimate. I'm happy to start with a number that the group is aligned on, maybe on the call today we can briefly discuss what that is. @robert-zaremba had suggested that we allow for optional extension by committee explicitly, and my feeling is that if we do that, we should set the standard deadline to be shorter than 8 weeks. Better to start with tight expectations and loosen them if we cannot hold. I thought 4 weeks seemed reasonable as a test of our ability to shift to more rapid and iterative review process. Would 6 be a better compromise? 3 rounds (6 weeks) of in person discussion seems like a reasonable expected time for total resolution or most ADRs.

We can always start with something shorter and update it to longer if necessary. I worry that if we start with a standard time bound that is too relaxed long, it won't actually improve our process at all and we won't take action to update the number to be shorter.

Reply:
Sorry, it's my fault. I've not made myself clear. I meant that the following:

```
1 week for initial review by reviewers
2 weeks to merge the PR (in draft / proposed state is fine)
4 weeks to conclusion of ADR
```

did not seem to be aligned with the original guesstimate of 8 weeks. I think that 8 weeks is a reasonable deadline to start with.

If that works well (and I believe it will), we could always shrink it down to 7 or 6 weeks eventually. I'd avoid setting the bar too high from the very beginning. Having this process in place is a _huge improvement per se_ compared to the current workflow of _not having a process at all_.

I'm suggesting to allow committee an extra week for the initial review exactly because `we have a 2 week cycle for architecture review calls`. We can take it from there and improve iteratively.

Reply:
I'd be in favor of seeing the initial review process at 1 week if most others think it's possible. If it was 2 weeks that means that it's much less likely to be reviewed without overlapping a bi-weekly review call. Furthermore in general it seems like it's better to try being ambitious and dial back when it's seen not to be feasible compared to the other way around.